File: solutions/_1897_redistribute_characters_to_make_all_strings_equal/solution.cpp
#include <string>
#include <vector>
using namespace std;

class Solution
{
public:
    bool makeEqual(vector<string> &words)
    {
        int n = words.size();
        vector<int> freq(26, 0);
        for (string &word : words)
        {
            for (char c : word)
            {
                freq[c - 'a']++;
            }
        }
        for (int f : freq)
        {
            if (f % n != 0)
            {
                return false;
            }
        }
        return true;
    }
};

/*
https://leetcode.com/problems/redistribute-characters-to-make-all-strings-equal/
*/

File: solutions/_1897_redistribute_characters_to_make_all_strings_equal/solution.py
from typing import List
from collections import Counter


class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        n = len(words)
        ctr = Counter()
        for word in words:
            ctr.update(word)
        for _, value in ctr.items():
            if value % n != 0:
                return False
        return True


# https://leetcode.com/problems/redistribute-characters-to-make-all-strings-equal/
